src/macros.rs

```diff
@@ -327,3 +327,148 @@ macro_rules! template {
+///   the `RenderOnce` trait (e.g. `String` or `&str`).
+///
+/// * `label_expression => cond_test` -- Conditionally include `label_expression` whenever `cond_test` is `true`.
+///   `cond_test` is an expression that returns either `true` or `false`.
+///
+/// This useful in generating class attribute as follows:
+///
+/// ```norun
+/// html! {
+///     div(class = labels!("active" => true, "button-style")) {
+///         : "This is a button"
+///     }
+/// }
+/// ```
+///
+#[macro_export]
+macro_rules! labels {
+
+    ($($tail:tt)+) => (
+        labels_sep_by!(" "; $($tail)*)
+    );
+
+}
+
+/// Utility macro for conditionally joining labels, such as `String` or `&str`,
+/// or anything that implements the `RenderOnce` trait can be used.
+///
+///
+/// Usage: `labels_sep_by!(seperator; maybe_label,...)`
+///
+/// * `seperator` -- Delimiter/seperator that implements the `RenderOnce` trait (e.g. `String` or `&str`).
+///
+/// * `maybe_label` -- Either `label_expression`, or `label_expression => cond_test`.
+///
+/// * `label_expression` -- An expression that returns a label that implements
+///   the `RenderOnce` trait (e.g. `String` or `&str`).
+///
+/// * `label_expression => cond_test` -- Conditionally include `label_expression` whenever `cond_test` is `true`.
+///   `cond_test` is an expression that returns either `true` or `false`.
+///
+/// This useful in generating style attribute as follows:
+///
+/// ```norun
+/// html! {
+///     div(style = labels_sep_by!(";"; "color: #000" => true, "font-weight: bold")) {
+///         : "This is a button"
+///     }
+/// }
+/// ```
+///
+#[macro_export]
+macro_rules! labels_sep_by {
+
+    (@inner_expand $has_before:expr; $sep:expr; $tmpl:ident $item:expr) => {
+        if $has_before {
+            $crate::RenderOnce::render_once($sep, $tmpl);
+        }
+        $crate::RenderOnce::render_once($item, $tmpl);
+    };
+
+    (@inner_expand $has_before:expr; $sep:expr; $tmpl:ident $item:expr => $should_include:expr) => {
+        if $should_include {
+            if $has_before {
+                $crate::RenderOnce::render_once($sep, $tmpl);
+            }
+            $crate::RenderOnce::render_once($item, $tmpl);
+        }
+    };
+
+    (@inner_expand $has_before:expr; $sep:expr; $tmpl:ident $item:expr, $($tail:tt)+) => {
+        if $has_before {
+            $crate::RenderOnce::render_once($sep, $tmpl);
+        }
+        $crate::RenderOnce::render_once($item, $tmpl);
+        labels_sep_by!(@inner_expand true; $sep; $tmpl $($tail)*);
+    };
+
+    (@inner_expand $has_before:expr; $sep:expr; $tmpl:ident $item:expr => $should_include:expr, $($tail:tt)+) => {
+        if $should_include {
+            if $has_before {
+                $crate::RenderOnce::render_once($sep, $tmpl);
+            }
+            $crate::RenderOnce::render_once($item, $tmpl);
+            labels_sep_by!(@inner_expand true; $sep; $tmpl $($tail)*);
+        } else {
```

Comment:
This will generate a lot of code. It may optimize away but will likely hurt compile times (not like the rest of horrorshow helps here...). Maybe just call the macro once and pass `$should_include || $has_before`?